### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by dan-holmes**

### CODE REVIEW

File: src/NewPostForm.js [hunk 1]
```diff
@@ -8,47 +8,44 @@ class NewPostForm extends React.Component {
...
 
   handleChange(event) {
     this.setState({ message: event.target.value });
   }
 
-  handleSubmit(event) {
-    event.preventDefault();
-    let data = {
-      message: this.state.message
+  fetchParams() {
```
Comment (arabellaknowles) [inline src/NewPostForm.js]: this could be a private function

File: src/App.js [hunk 2]
```diff
@@ -101,39 +92,43 @@ class App extends React.Component {
     })
   }
 
-  handleClick(e) {
+  loadHomePage() {
```
Comment (arabellaknowles) [inline src/App.js:95]: does this need a prevent default function?

File: src/Post.js [hunk 3]
```diff
@@ -37,23 +35,32 @@ class Post extends React.Component {
...
-    })
-    this.show()
+    this.setState({ editing: true })
   }
 
-  handleSubmit(event) {
-    event.preventDefault();
-    let data = {
-      message: this.state.updated_post
+  fetchParams() {
```
Comment (arabellaknowles) [inline src/Post.js]: this could be a private function

Comment (arabellaknowles) [reply src/Post.js]: also basically the same as new post fetchParams function, is there a way to just have one function and pass argument of the value of message: in data?

File: src/Post.js [hunk 4]
```diff
@@ -37,23 +35,32 @@ class Post extends React.Component {
...
+      credentials: 'include',
+      body: JSON.stringify(data)
     }
-    let current_post_id = this.state.post_id
-    updateMessage(data, current_post_id)
+  }
+
+  submitEdit(event) {
+    event.preventDefault();
+    let httpRequest = `${BASE_URL}/api/v1/posts/` + this.props.data.id
```
Comment (arabellaknowles) [inline src/Post.js]: you have called this.props.data.id in the state so would only need this.state.post_id

File: src/Post.js [hunk 5]
```diff
@@ -63,35 +70,12 @@ class Post extends React.Component {
...
-      .catch(err => console.log(err))
-  }
-
   render() {
     let edit_button = '';
     let delete_button = '';
-    // const post_id = this.props.data.id;
-    let edit_link = "#posts/edit/" + this.state.post_id
 
     if (this.props.data.editable && this.props.user.authCompleted) {
```
Comment (arabellaknowles) [inline src/Post.js]: if you're in the mindset of refactoring both this.props.data.editable, owned by and authCompleted could be defined at the beginning of render like done in app.js

File: src/PostList.js [hunk 6]
```diff
@@ -20,44 +21,53 @@ class PostList extends React.Component {
...
-    this.setState({
-      isLoaded: value
-    })
-  }
-
   render() {
-    const { error, isLoaded, posts } = this.state;
+    const { error, loadedFor, posts } = this.state;
+    console.log(this.state.loadedFor);
+    console.log(this.props.user.current.id);
```
Comment (arabellaknowles) [inline src/PostList.js]: these can be deleted?

File: src/PostList.js [hunk 7]
```diff
@@ -20,44 +21,53 @@ class PostList extends React.Component {
...
+    if (this.state.loadedFor !== this.props.user.current.id && !this.state.awaitingResults) {
```
Comment (arabellaknowles) [inline src/PostList.js]: this statement is quite difficult to wrap your head around, maybe refactor for readability

Comment (arabellaknowles) [reply src/PostList.js]: is it carried out like this: ((this.state.loadedFor !== this.props.user.current.id) && !this.state.awaitingResults)

File: src/SignUpForm.js [hunk 8]
```diff
@@ -24,56 +24,48 @@ class signUpForm extends React.Component {
...
+      body: JSON.stringify(data)
+    })
+  }
+
+  signUp(event) {
+    event.preventDefault();
+    fetch(`${BASE_URL}/api/v1/users`, this.fetchParams())
+      .then(res => res.json())
+      .then(res => this.props.setUserData(res))
+      .catch(err => alert('Error with sign up'))
```
Comment (arabellaknowles) [inline src/SignUpForm.js]: future changes: actually return the error the backend outputs

**Review by arabellaknowles — APPROVED** [review]
WOW!! MIND BLOWN!! Some absolutely incredible refactoring!!! - I would change all the fetchParams() functions to private methods :)

File: src/NewPostForm.js [hunk 9]
```diff
@@ -8,47 +8,44 @@ class NewPostForm extends React.Component {
...
+    let data = { message: this.state.message }
```
Comment (arabellaknowles) [inline src/NewPostForm.js:26]: where is data being reused

File: src/App.js (hunk 2 above)
Comment (dan-holmes) [reply src/App.js:95]: It does not, it's called in the header along side a prevent default

File: src/Post.js (hunk 3 above)
Comment (dan-holmes) [reply src/Post.js]: That probably is the dryest way, but there's enough changes that it gets pretty fiddly. method, credentials, body vs no body, cors. I think you're right but I'm going to leave it as is for now

File: src/SignUpForm.js (hunk 8 above)
Comment (dan-holmes) [reply src/SignUpForm.js]: Agreed

File: src/PostList.js (hunk 7 above)
Comment (dan-holmes) [reply src/PostList.js]: I have now tidied this up!

**Pull request merged**